dataflux_pytorch/lightning/dataflux_lightning_checkpoint.py

```diff
@@ -0,0 +1,77 @@
+import torch
+from typing import Optional, Dict, Any
+
+from lightning.pytorch.plugins.io import CheckpointIO
+from google.cloud import storage
+from google.api_core.client_info import ClientInfo
+
+class DatafluxLightningCheckpoint(CheckpointIO):
+    """A checkpoint manager for GCS using the :ckass:'CheckpointIO' interface"""
+
+    def __init__(
+        self,
+        project_name: str,
+        bucket_name: str,
+        storage_client: Optional[storage.Client] = None,
+    ):
+        self.project_name = project_name
+        self.bucket_name = bucket_name
+        self.storage_client = storage_client
+        if not storage_client:
+            self.storage_client = storage.Client(
+                project=self.project_name,
+                client_info=ClientInfo(user_agent="dataflux/0.0"),
+            )
+        self.bucket = self.storage_client.bucket(self.bucket_name)
+
+    def _parse_gcs_path(self, path: str) -> str:
+        if not path or not path.startswith("gcs://"):
+            raise ValueError("Path needs to begin with gcs://")
+        path = path[len("gcs://") :]
```

Comment:
Oh I just realized that we are using `gcs://` as the prefix. I feel that this is okay for now but just a heads up that gcsfs and GCS APIs (such as gsutil and gcloud cp, etc) are paths starting with `gs://`. Not sure if we want to make the conventions uniform

Reply:
I think [gcsfs](https://gcsfs.readthedocs.io/en/latest/#integration) uses `gcs://` as well. Lightning documentation also confirms this https://lightning.ai/docs/pytorch/stable/common/remote_fs.html 

If there is an edit I think it is to support `gs://` paths here, I can go ahead and add that as an allowable prefix?

Reply:
Oh right. It supports both `gcs://` and `gs://`. Yeah it would be great to allow this flexibility. Thanks!